newsroom/companies/views.py

```diff
@@ -127,7 +127,10 @@ def delete(_id):
     """
     Deletes the company and users of the company with given company id
     """
-    get_resource_service('users').delete_action(lookup={'company': ObjectId(_id)})
+    try:
+        get_resource_service('users').delete_action(lookup={'company': ObjectId(_id)})
+    except Exception:
+        return jsonify({'company': gettext('Can not delete current user.')}), 403
```

Comment:
It would be better to provide this error message in the `UsersService.on_delete` method, which would allow this `try...except` to catch all types of errors and pass them onto the user.

Because of this change, it would also make sense to use a more generic key for the error instead of `company`, maybe `error`.

Reply:
I have added the error message here so as to provide same error message to the UI.